frontend/src/app/util/text/content/to_dom.cljs

```diff
@@ -0,0 +1,129 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) KALEIDOS INC
+
+(ns app.util.text.content.to-dom
+  (:require
+   [app.common.data :as d]
+   [app.common.text :as txt]
+   [app.util.dom :as dom]
+   [app.util.text.content.styles :as styles]))
+
+(defn set-dataset
+  [element data]
+  (doseq [[data-name data-value] data]
+    (dom/set-data! element (name data-name) data-value)))
+
+(defn set-styles
+  [element styles]
+  (doseq [[style-name style-value] styles]
+    (if (contains? styles/mapping style-name)
+      (let [[style-encode] (get styles/mapping style-name)
+            style-encoded-value (style-encode style-value)]
+        (dom/set-style! element (styles/get-style-name-as-css-variable style-name) style-encoded-value))
+      (dom/set-style! element (styles/get-style-name style-name) (styles/normalize-style-value style-name style-value)))))
+
+(defn create-element
+  ([tag]
+   (create-element tag nil nil))
+  ([tag attrs]
+   (create-element tag attrs nil))
+  ([tag attrs children]
+   (let [element (dom/create-element tag)]
+     ;; set attributes to the element if necessary.
+     (when (some? attrs)
+       (doseq [[attr-name attr-value] attrs]
+         (cond
+           (= attr-name :data)
+           (set-dataset element attr-value)
+           (= attr-name :style)
+           (set-styles element attr-value)
+           :else
+           (dom/set-attribute! element (name attr-name) attr-value))))
+
+     ;; add childs to the element if necessary.
+     (when (some? children)
```

Comment:
You don't need to check for some? `nil` is iterable. So you can iterate directly over children and over attrs.